docker/base.Dockerfile

```diff
@@ -1,6 +1,6 @@
 # syntax=docker/dockerfile:1
 # Build the manager binary
-FROM --platform=$BUILDPLATFORM mcr.microsoft.com/oss/go/microsoft/golang:1.24.6@sha256:aad3eb77b1d422f5b11ebf8e9bef04146ff7206ab8c319bab33078a27746e500 AS builder 
+FROM --platform=$BUILDPLATFORM golang:1.25.0 AS builder
```

Comment:
we need to use the mcr version - if you take a look at the closed PR from dependabot it should have a sha you can use https://github.com/Azure/kube-egress-gateway/pull/1160